examples/mp-membership/src/main.rs

```diff
@@ -613,6 +635,128 @@ fn test_scalability_crash(config: &TestConfig) -> TestResult {
     Ok(true)
 }
 
+// MODEL CHECKING ------------------------------------------------------------------------------------------------------
+
+fn mc_invariant_check_stabilized(group: Vec<String>) -> InvariantFn {
+    boxed!(move |state| {
+        let group = group.clone().into_iter().collect::<HashSet<String>>();
+        for node in state.node_states.keys() {
+            if let Some(msg) = state.node_states[node].proc_states[node].local_outbox.first() {
+                if msg.tip != "MEMBERS" {
+                    return Err("wrong message type".to_owned());
+                }
+                let data: MembersMessage = serde_json::from_str(&msg.data).unwrap();
+                let members = HashSet::from_iter(data.members.into_iter());
+                if !members.eq(&group) {
+                    return Err(format!("expected a stabilized group {:?} but got {:?}", group, members));
+                }
+            }
+        }
+        Ok(())
+    })
+}
+
+fn mc_explore_after_joins(sys: &mut System, seed_proc: String) -> Result<HashSet<McState>, String> {
+    let procs = sys.process_names();
+    let mut mc = ModelChecker::new(sys);
+
+    let strategy_config = StrategyConfig::default()
+        // Every process fires no more than two timers
+        // so we expect each node to initialize communication with others at most 3 times:
+        // 1 times on a local message and 2 times on a timer
+        .prune(prunes::events_limit_per_proc(
+            |entry: &LogEntry, proc_name: &String| match entry {
+                LogEntry::McTimerFired { proc, .. } => proc_name == proc,
+                _ => false,
+            },
+            procs.clone(),
+            2,
+        ))
+        // We can explore about 20 steps of simulation
+        .goal(goals::any_goal(vec![goals::no_events(), goals::depth_reached(20)]))
+        // And 2 minutes is more than enough for this.
+        .invariant(invariants::time_limit(Duration::from_secs(120)))
+        // We check for states in which group should be stabilized:
+        // Either nothing is going to change or every process received at least 3 messages.
+        // Considering we have 3 nodes in the system and network is stable, we can show this is enough:
+        // * seed node (A) should get information from both _side_ nodes (B & C)
+        // * because one seed node joins earlier than another, the later will know full group as well
+        // * the early side node is able to get information about group either once again from seed,
+        // or directly from other seed, depending on implementation
+        // Technically, there exists a trace A <-> B (x3), A <-> C(x3) where B information becomes outdated,
+        // but we consider solution wrong if it can enter such trace.
```

Comment:
```suggestion
        // Collect states in which the group should be stabilized, namely:
        // either no events left, or every process received at least 3 messages.
        //
        // Considering a system with 3 processes and a stable network, we can show this is enough:
        // * seed process (A) should get information from both other processes (B & C)
        // * because C joins later than B, it will know the full group as well
        // * B is able to get information about the full group either from A or C, depending on implementation
        //
        // Technically, there can be execution A <-> B (x3), A <-> C (x3) where B's information becomes outdated,
        // but we consider solution wrong if it allows such execution.
```